src/black/mode.py

```diff
@@ -183,5 +184,6 @@ def get_cache_key(self) -> str:
             str(int(self.is_ipynb)),
             str(int(self.magic_trailing_comma)),
             str(int(self.experimental_string_processing)),
+            ",".join(self.python_cell_magics),
```

Comment:
```suggestion
            ",".join(sorted(self.python_cell_magics)),
```
Set iteration order is randomized, so this would make the cache key unstable